### ISSUE TITLE
Empty line after module header?

### ISSUE THREAD

**Issue by Kosmas** [niftyn8/elixir_style_guide#36]
Any suggestions about having/not having an empty line between module header and first function?
And could we add it to the guide?

So which one should be preferred?

defmodule some_module do

  def some_function do
    # function body
  end
end

or 

defmodule some_module do
  def some_function do
    # function body
  end
end

**Comment by knewter** [comment]
I tend to not have a separator between top-of-module and first-function, but I *do* have a separator between "alias, use, etc statements" and first-function.  So I think my rule of thumb goes something like:

- No newline after defmodule
- No newline before first function def
- Newline after "module-level-code-blocks"

**Comment by Kosmas** [comment]
@knewter sounds reasonable.

I've just done a PR with your suggestions: https://github.com/niftyn8/elixir_style_guide/pull/37

**Comment by asaaki** [comment]
I tend to use the same style like @knewter.

```elixir
defmodule Foo do
  use SomethingElse
  alias Another.FancyModule

  def my_func do
    # ...
  end

  def another_func(true), do: 1
  def another_func(_any), do: 2

  def bar do
    # ...
  end
end
```

Basically reduced down to: Add empty lines between (different) functions and function calls/keywords. Don't do put a newline in the beginning or end of a module block and the first/last statement.

I still do not put an empty line between function definitions with identical arity/signature; this is still an open issue at #31.

But this is also just what I adopted from the Ruby community and their style guide.
As both languages look pretty similar it's easy to do so.

**Comment by knewter** [comment]
Yeah, I think my biggest problem with taking my bullet list into the guide
verbatim is it's not example-y enough yet.

On Fri, Mar 6, 2015 at 10:28 AM, Christoph Grabo <notifications@github.com>
wrote:

> I tend to use the same style like @knewter <https://github.com/knewter>.
>
> defmodule Foo do
>   use SomethingElse
>   alias Another.FancyModule
>
>   def my_func do    # ...
>   end
>
>   def another_func(true), do: 1
>   def another_func(_any), do: 2
>
>   def bar do    # ...
>   endend
>
> Basically reduced down to: Add empty lines between (different) functions
> and function calls/keywords. Don't do put a newline in the beginning or end
> of a module block and the first/last statement.
>
> I still do not put an empty line between function definitions with
> identical arity/signature; this is still an open issue at #31
> <https://github.com/niftyn8/elixir_style_guide/issues/31>.
>
> But this is also just what I adopted from the Ruby community and their
> style guide.
> As both languages look pretty similar it's easy to do so.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/niftyn8/elixir_style_guide/issues/36#issuecomment-77588078>
> .
>



-- 
*Josh Adams*
CTO | isotope|eleven <http://www.isotope11.com>
[cell] (205) 215-3957
[work] (877) 476-8671 x201

**Comment by asaaki** [comment]
I guess this issue can be closed, as the referenced PR #37 is already merged.
cc @niftyn8, @knewter
